ext/opcache/Optimizer/block_pass.c

```diff
@@ -115,6 +115,10 @@ static int get_const_switch_target(zend_cfg *cfg, zend_op_array *op_array, zend_
 		/* fallback to next block */
 		return block->successors[block->successors_count - 1];
 	}
+	if (opline->opcode == ZEND_MATCH && Z_TYPE_P(val) != IS_LONG && Z_TYPE_P(val) != IS_STRING) {
+		/* always jump to the default arm */
+		return block->successors[block->successors_count - 2];
```

Comment:
When removing the  fall-through successor, you'll of course also have to adjust this to -1. Possibly that's the issue you hit?

Reply:
That was indeed the issue, thanks!